### PR TITLE
Quic ietf 4967 v18

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/4967
https://redmine.openinfosecfoundation.org/issues/5143
https://redmine.openinfosecfoundation.org/issues/5166

Describe changes:
- Parses standardized QUIC IETF v1
- Parses gquic version Q039, 0xfaceb002, etc...
- JA3 logging and detection on quic
- Adds quic events and rules 
- bump up rust carte tls_parser already used by rep, so that it can be used by quic
- Rustfmt previous code

suricata-verify-pr: 783
https://github.com/OISF/suricata-verify/pull/783

Replaces https://github.com/OISF/suricata/pull/7606 with rebase and last commit to reassemble multiple out of order crypto frames in order to be able to parse the tls hello in it

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 8425

File: rust/src/quic/frames.rs [hunk 1]
```diff
@@ -210,6 +444,28 @@ fn parse_stream_frame(input: &[u8], frame_ty: u8) -> IResult<&[u8], Frame, QuicE
     ))
 }
 
+fn parse_crypto_stream_frame(input: &[u8]) -> IResult<&[u8], Frame, QuicError> {
+    let (rest, _offset) = quic_var_uint(input)?;
+    let (rest, data_length) = quic_var_uint(rest)?;
+    let (rest, stream_data) = take(data_length)(rest)?;
```
Comment (victorjulien) [inline rust/src/quic/frames.rs:450]: On my ARM 32 bit builder I get an error here:
```
   Compiling suricata v7.0.0-dev (/builds/inliniac/suricata-ci/suricata/rust)
error[E0277]: the trait bound `u64: nom::ToUsize` is not satisfied
   --> src/quic/frames.rs:450:36
    |
450 |     let (rest, stream_data) = take(data_length)(rest)?;
    |                               ---- ^^^^^^^^^^^ the trait `nom::ToUsize` is not implemented for `u64`
    |                               |
    |                               required by a bound introduced by this call
    |
    = help: the following other types implement trait `nom::ToUsize`:
              u16
              u32
              u8
              usize
note: required by a bound in `nom::bytes::complete::take`
   --> /builds/inliniac/suricata-ci/suricata/rust/vendor/nom-5.1.2/src/bytes/complete.rs:381:6
    |
381 |   C: ToUsize,
    |      ^^^^^^^ required by this bound in `nom::bytes::complete::take`
```
`usize` is 32 bit on 32 bit platforms, so the question here is what to do about the case where `data_length` exceeds the max 32 value.

Comment (victorjulien) [reply rust/src/quic/frames.rs:450]: As an aside: I'm surprised to see nom 5 in this error. I would expect new code to be all nom 7.

Comment (catenacyber) [reply rust/src/quic/frames.rs:450]: > As an aside: I'm surprised to see nom 5 in this error. I would expect new code to be all nom 7.

I think this is not an aside, this is the main problem, or the root cause of it...

Comment (catenacyber) [reply rust/src/quic/frames.rs:450]: @ct0br0 would we have a 32 bit build in Github CI ?

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/7676

File: rust/src/quic/frames.rs (hunk 1 above)
Comment (catenacyber) [reply rust/src/quic/frames.rs:450]: @victorjulien next version should fix these

**Pull request closed without merging**